crypto/src/main/java/org/web3j/crypto/StructuredDataEncoder.java

```diff
@@ -109,6 +215,8 @@ public static String encodeType(String primaryType) {
         encTypes.add("bytes32");
         encValues.add(typeHash(primaryType));
 
+        String arrayTypeRegex = "^([a-zA-Z_$][a-zA-Z_$0-9]*)((\\[([1-9]\\d*)?\\])+)$";
```

Comment:
I guess if regexp works it works but if it doesn't somebody else would spend some time decoding it. I would at least  add a comment explaining what it matches with some examples.